app/models/course.rb

```diff
@@ -53,6 +52,8 @@ class Course < ActiveRecord::Base
                                     dependent: :destroy
   has_many :lesson_plan_events, through: :lesson_plan_items,
                                 source: :actable, source_type: Course::LessonPlan::Event.name
+  # Achievements must be declared after material_folders or duplication will fail.
+  has_many :achievements, dependent: :destroy
```

Comment:
The only link I can come up with is that `Achievements` have `Conditions` that are `Assessments` which `has_one_folder`. Is that the link on why folders would need to be declared first?